elasticsearch/shard_migration.py

```diff
@@ -0,0 +1,181 @@
+        return []
+
+# Move all shards for an index from a source node to one of the target nodes
+def move_shards(indices, source_node, target_nodes, es_host):
+    for index_data in indices:
+        index = index_data["index"]
+        shard = index_data["shard"]
+
+        for target_node in target_nodes:
+            reroute_body = {
+                "commands": [
+                    {
+                        "move": {
+                            "index": index,
+                            "shard": shard,
+                            "from_node": source_node,
+                            "to_node": target_node,
+                        }
+                    }
+                ]
+            }
+
+            try:
+                LOG.info(
+                    f"Trying to move shard {shard} of index '{index}' from {source_node} to {target_node}."
+                )
+                headers = {"Content-Type": "application/json"}
+                response = requests.post(
+                    f"{es_host}/_cluster/reroute",
+                    headers=headers,
+                    data=json.dumps(reroute_body),
+                )
+                response.raise_for_status()
+                LOG.info(
+                    f"Shard {shard} of index '{index}' successfully moved from {source_node} to {target_node}."
+                )
+                return True
+            except requests.exceptions.RequestException as e:
+                LOG.warning(
+                    f"Not able to move shard {shard} of index '{index}' to {target_node}. Error: {e}"
+                )
+
+        LOG.warning(
+            f"Shard {shard} of index '{index}' could not be moved to any of the target nodes in the list."
+        )
+
+
+def filter_new_shard_allocation(source_node_ip, es_host):
+    try:
+        LOG.info(f"Filtering new shard allocation to on source node: {source_node_ip}.")
+
+        body = {
+            "persistent": {"cluster.routing.allocation.exclude._ip": source_node_ip}
+        }
+
+        headers = {"Content-Type": "application/json"}
+        response = requests.put(
+            f"{es_host}/_cluster/settings", headers=headers, data=json.dumps(body)
+        )
+        response.raise_for_status()
+        LOG.info(
+            f"Successfully updated cluster settings for filtering new shard allocation on source node {source_node_ip}."
+        )
+    except requests.exceptions.RequestException as e:
+        LOG.error(
+            f"Error updating cluster settings for filtering new shard allocation on source node {source_node_ip}: {e}"
+        )
+
+
+def main():
+    (opts, args) = parse_opts()
+
+    if not opts.source_node or not opts.source_node_ip or not opts.target_nodes:
+        LOG.error(
+            "Options --source-node, --source-node-ip, and --target-nodes are required."
+        )
+        exit(2)
+
+    es_host = opts.es_host
+    source_node = opts.source_node
+    source_node_ip = opts.source_node_ip
```

Comment:
> This could be resolved by simply adding `.wg` to `source_node`. The flag could still be useful but it can be optional.

When I try to do it by adding `.wg` to the source_node:
```bash
mburcul@store-02.he-eu-hel1.logs.hq:~ % curl -X PUT "localhost:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
{
  "persistent" : {
    "cluster.routing.allocation.exclude._ip" : "store-01.he-eu-hel1.logs.hq.wg"
  }
}
'

{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "invalid IP address [store-01.he-eu-hel1.logs.hq.wg] for [cluster.routing.allocation.exclude._ip]"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "invalid IP address [store-01.he-eu-hel1.logs.hq.wg] for [cluster.routing.allocation.exclude._ip]"
  },
  "status" : 400
}
```

Reply:
> This could be resolved by simply adding `.wg` to `source_node`. The flag could still be useful but it can be optional.

When I add `.wg` to the `source_node` I get:
```bash
mburcul@store-02.he-eu-hel1.logs.hq:~ % curl -X PUT "localhost:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
{
  "persistent" : {
    "cluster.routing.allocation.exclude._ip" : "store-02.he-eu-hel1.logs.hq.wg"
  }
}
'

{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "invalid IP address [store-02.he-eu-hel1.logs.hq.wg] for [cluster.routing.allocation.exclude._ip]"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "invalid IP address [store-02.he-eu-hel1.logs.hq.wg] for [cluster.routing.allocation.exclude._ip]"
  },
  "status" : 400
}
```

Reply:
You have to do the DNS lookup yourself.